schema/bom-1.4-strict-SNAPSHOT.schema.json

```diff
@@ -1074,6 +1082,278 @@
           "description": "The value of the property."
         }
       }
+    },
+    "advisory": {
+      "type": "object",
+      "title": "Advisory",
+      "description": "Location of additional advisory information.",
+      "required": ["url"],
+      "additionalProperties": false,
+      "properties": {
+        "title": {
+          "type": "string"
+        },
+        "url": {
+          "type": "string",
+          "format": "iri-reference"
+        }
+      }
+    },
+    "cwe": {
+      "type": "integer",
+      "minimum": 1,
+      "title": "CWE",
+      "description": "Integer representation of a Common Weaknesses Enumerations (CWE). For example 399 (of https://cwe.mitre.org/data/definitions/399.html)"
+    },
+    "severity": {
+      "type": "string",
+      "title": "Severity",
+      "description": "Textual representation of the severity of the vulnerability adopted by the risk analysis method. If an other risk analysis method is used other than whats defined in scoreSourceType, the user is expected to translate appropriately to match with an element value below.",
+      "enum": [
+        "critical",
+        "high",
+        "medium",
+        "low",
+        "none",
+        "unknown"
+      ]
+    },
+    "scoreMethod": {
+      "type": "string",
+      "title": "method",
+      "description": "Specifies the risk scoring methodology/standard used.",
+      "enum": [
+        "CVSSv2",
+        "CVSSv3",
+        "CVSSv31",
+        "OWASP",
+        "Other"
+      ]
+    },
+    "impactAnalysisState": {
+      "type": "string",
+      "title": "Impact Analysis State",
+      "description": "Textual representation of the state of the impact analysis.",
+      "enum": [
+        "exploitable",
+        "in_triage",
+        "false_positive",
+        "not_affected",
```

Comment:
Should we handle a few choices on why a vulnerability is `not_applicable`?

Examples: code unreachable, vulnerable code not included (eg. with webpack), etc.

Reply:
This sounds akin to consumption of the component, and slightly unrelated to a vulnerability existing in a component, or am I thinking about this wrong?

Reply:
Thats correct. This is a VEX use case. The essence is that not every component is going to be exploitable. Some components may never be called, or the vulnerable method or function is never called, etc. This is really about trying to communicate, as a software supplier, the inherited vulnerabilities as well as whether or not consumers will be impacted by them.

It also allows IAST/RASP vendors, researchers, etc, to communicate with certainty, whether or not a vulnerability was exploitable or not.

Reply:
@planetlevel if a vulnerability is not exploitable, I'm thinking about adding a 'justification' in the event the analysis is set to 'not_affected'.

The choices I'm thinking of are:
* code_not_reachable - the code cannot be reached at runtime
* code_not_present - the code has been removed, tree-shaked (e.g. via webpack)
* function_not_enabled - the code requires a configurable option that is not set
* compiled_by_avoidance - a compiler flag was set/unset which prevented the introduction of the vulnerability
* dependency_not_met  - the vulnerability requires the presence of another component which is not present in the system.

Do you agree with these?
Are there others?

Reply:
This is a really good list.  I think adding code_requires_environment, code_protected_at_runtime, and code_protected_at_perimeter would help.

My full (OCD-refactored) list would look like:
* code_not_present - the code has been removed, tree-shaked (e.g. via webpack)
* code_not_reachable - the vulnerable code is not invoked at runtime
* code_requires_configuration - exploitability requires a configurable option to be set/unset
* code_requires_dependency - exploitability requires a dependency that is not present
* code_requires_environment - exploitability requires a certain environment which is not present
* code_protected_by_compiler - exploitability requires a compiler flag to be set/unset
* code_protected_at_runtime - exploits are prevented at runtime
* code_protected_at_perimeter - attacks are blocked at network perimeter
* code_protected_by_mitigating_control - preventative measures have been implemented that reduce the likelihood and/or impact of the vulnerability

Reply:
Added justifications and descriptions in b7e37c6f7125d6c5ee2cf03323144fe81608c438